pandas/core/tools/datetimes.py

```diff
@@ -36,9 +36,66 @@ def _guess_datetime_format_for_array(arr, **kwargs):
         return _guess_datetime_format(arr[non_nan_elements[0]], **kwargs)
 
 
+def _maybe_cache(arg, format, cache, tz, _convert_listlike):
+    """
+    Create a cache of unique dates from an array of dates
+
+    Parameters
+    ----------
+    arg : integer, float, string, datetime, list, tuple, 1-d array of dates
+    format : string, strftime to parse time
```

Comment:
can you format as a doc-string, IOW types on the first line, 2nd line for what it is doing.